traffic_ops/traffic_ops_golang/deliveryservice/sslkeys.go

```diff
@@ -115,3 +118,70 @@ func generatePutRiakKeys(req tc.DeliveryServiceGenSSLKeysReq, tx *sql.Tx, tv tra
 	}
 	return nil
 }
+
+// GeneratePlaceholderSelfSignedCert generates a self-signed SSL certificate as a placeholder when a new HTTPS
+// delivery service is created or an HTTP delivery service is updated to use HTTPS.
+func GeneratePlaceholderSelfSignedCert(ds tc.DeliveryServiceV40, inf *api.APIInfo, context context.Context) (error, int) {
+	version := util.JSONIntStr(1)
+
+	db, err := api.GetDB(context)
+	if err != nil {
+		return err, http.StatusInternalServerError
+	}
+	tx, err := db.Begin()
+	if err != nil {
+		return err, http.StatusInternalServerError
+	}
+	defer tx.Commit()
+
+	cdnName, cdnDomain, err := getCDNNameDomain(*ds.CDNID, tx)
+	if err != nil {
+		return err, http.StatusInternalServerError
+	}
+
+	cdnNameStr := string(cdnName)
+
+	if ds.ExampleURLs == nil {
+		ds.ExampleURLs = MakeExampleURLs(ds.Protocol, *ds.Type, *ds.RoutingName, *ds.MatchList, cdnDomain)
+	}
+
+	hostname := strings.Split(ds.ExampleURLs[0], "://")[1]
+	if strings.Contains(ds.Type.String(), "HTTP") {
+		parts := strings.Split(hostname, ".")
+		parts[0] = "*"
+		hostname = strings.Join(parts, ".")
+	}
+
+	req := tc.DeliveryServiceGenSSLKeysReq{
+		DeliveryServiceSSLKeysReq: tc.DeliveryServiceSSLKeysReq{
+			CDN:             &cdnNameStr,
+			DeliveryService: ds.XMLID,
+			HostName:        &hostname,
+			Key:             ds.XMLID,
+			Version:         &version,
+			BusinessUnit:    util.StrPtr("Placeholder"),
+			City:            util.StrPtr("Placeholder"),
+			Organization:    util.StrPtr("Placeholder"),
+			Country:         util.StrPtr("United States (US)"),
```

Comment:
Does "Placeholder" not work for country and state? If not, disregard :)

Reply:
haha i think it will work but it felt weird. updated now